internal/provider/resources/resource_team_roles.go

```diff
@@ -0,0 +1,329 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/samber/lo"
+	"net/http"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients"
+	"github.com/astronomer/terraform-provider-astro/internal/clients/iam"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/models"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/schemas"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &teamRolesResource{}
+var _ resource.ResourceWithImportState = &teamRolesResource{}
+var _ resource.ResourceWithConfigure = &teamRolesResource{}
+
+func NewTeamRolesResource() resource.Resource {
+	return &teamRolesResource{}
+}
+
+// teamRolesResource defines the resource implementation.
+type teamRolesResource struct {
+	iamClient      *iam.ClientWithResponses
+	organizationId string
+}
+
+func (r *teamRolesResource) Metadata(
+	ctx context.Context,
+	req resource.MetadataRequest,
+	resp *resource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_team_roles"
+}
+
+func (r *teamRolesResource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Team Roles resource",
+		Attributes:          schemas.ResourceTeamRolesSchemaAttributes(),
+	}
+}
+
+func (r *teamRolesResource) Configure(
+	ctx context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	apiClients, ok := req.ProviderData.(models.ApiClientsModel)
+	if !ok {
+		utils.ResourceApiClientConfigureError(ctx, req, resp)
+		return
+	}
+
+	r.iamClient = apiClients.IamClient
+	r.organizationId = apiClients.OrganizationId
+}
+
+func (r *teamRolesResource) MutateRoles(
+	ctx context.Context,
+	data *models.TeamRoles,
+) diag.Diagnostics {
+	teamId := data.TeamId.ValueString()
+
+	// Then convert the models to the request types for the API
+	workspaceRoles, diags := RequestWorkspaceRoles(ctx, data.WorkspaceRoles)
+	if diags.HasError() {
+		return diags
+	}
+	deploymentRoles, diags := RequestDeploymentRoles(ctx, data.DeploymentRoles)
+	if diags.HasError() {
+		return diags
+	}
+
+	// create request
+	updateTeamRolesRequest := iam.UpdateTeamRolesJSONRequestBody{
```

Comment:
can you explain why this is an update and not a create? 

(i'm assuming it's because team roles are initialized to nil/empty?)

Reply:
for modifying team roles, we only have one endpoint in v1beta1 `UpdateTeamRoles` so we have to use that endpoint for all 3 mutations